src/main/java/org/jabref/logic/exporter/CffExporter.java

```diff
@@ -0,0 +1,227 @@
+package org.jabref.logic.exporter;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.time.Year;
+import java.time.YearMonth;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.jabref.logic.util.StandardFileType;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.Author;
+import org.jabref.model.entry.AuthorList;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.BiblatexSoftwareField;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.field.UnknownField;
+import org.jabref.model.entry.types.EntryType;
+import org.jabref.model.entry.types.StandardEntryType;
+
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * Exporter for exporting in CFF format.
+ */
+class CffExporter extends Exporter {
+    public CffExporter() {
+        super("cff", "CFF", StandardFileType.CFF);
+    }
+
+    @Override
+    public void export(BibDatabaseContext databaseContext, Path file, List<BibEntry> entries) throws Exception {
+        Objects.requireNonNull(databaseContext);
+        Objects.requireNonNull(file);
+        Objects.requireNonNull(entries);
+
+        if (entries.isEmpty()) { // Do not export if no entries to export -- avoids exports with only template text
+            return;
+        }
+
+        try (FileWriter writer = new FileWriter(file.toFile(), StandardCharsets.UTF_8)) {
+            DumperOptions options = new DumperOptions();
+
+            // Set line width to infinity to avoid line wrapping
+            options.setWidth(Integer.MAX_VALUE);
+
+            // Set collections to be written in block rather than inline
+            options.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK);
+            options.setPrettyFlow(true);
+
+            // Set indent for sequences to two spaces
+            options.setIndentWithIndicator(true);
+            options.setIndicatorIndent(2);
+            Yaml yaml = new Yaml(options);
+
+            Map<String, Object> originalData = new LinkedHashMap<>();
+            Map<String, Object> preferredData = new LinkedHashMap<>();
+            Map<String, Object> data = originalData;
+            data.put("cff-version", "1.2.0");
+
+            for (BibEntry entry : entries) {
+                // Retrieve all fields
+                Map<Field, String> entryMap = entry.getFieldMap();
```

Comment:
Better do a copy of the map. Not sure if the entry is changed when you change `entryMap`  below.

Reply:
I made a shallow copy of the map, which is sufficient since we only remove fields without altering any of them.

Reply:
Remove means that the map is changed. `getFieldMap` returns the internal map of the BibEntry. Thus, if you remove a field from there, the field is removed.

You can write a test case for `BibEntry` if you want to proof me wrong.

Reply:
I added the following test case in `BibEntryTest`, which passes. 
```java
@Test
    void shallowCopiedBibEntryDoesNotRemoveFields() throws Exception {
        entry.setField(StandardField.URL, "value");
        Map<Field, String> entryMap = entry.getFieldMap();
        Map<Field, String> entryMapCopy = new HashMap<>(entryMap);
        entryMapCopy.remove(StandardField.URL);
        assertEquals(entryMapCopy.size(), 0);
        assertEquals(entryMap.get(StandardField.URL), "value");
    }
```